code/modules/mob/living/simple_animal/friendly/rat.dm

```diff
@@ -43,10 +52,74 @@
 	response_harm = "stamps on"
 	holder_type = /obj/item/holder/rat/white/Milky
 
-/mob/living/simple_animal/mouse/rat/brown/Old_Timmy
-	name = "Old Timmy"
-	desc = "An ancient looking rat from the old days of the colony."
-	response_help = "pets"
-	response_disarm = "gently pushes aside"
-	response_harm = "splats"
-	holder_type = /obj/item/holder/rat/brown/Old_Timmy
+
+//Specific Pets for Frozen's Rat Collecting Competition
+
+/mob/living/simple_animal/mouse/rat/pet
+	name = "Pet Rat"
+	desc = "This is someone's pet rat. I wonder what it's doing here."
+	holder_type = /obj/item/holder/rat/pet
+
+/mob/living/simple_animal/mouse/rat/pet/marvin
+	name = "Marvin"
+	desc = "A sleek well kept rat with a tiny collar around its neck, it must belong to someone. For a rodent it appears remarkably clean and hygenic."
+	body_color = "black"
+	icon_state = "rat_black"
+	holder_type = /obj/item/holder/rat/pet/marvin
+
+/mob/living/simple_animal/mouse/rat/pet/ikit
+	name = "Ikit"
+	desc = "An albino rat with a tiny collar around its neck, it must belong to someone. Hope it doesn't have some sort of genetically engineered disease or something..."
+	body_color = "white"
+	icon_state = "rat_white"
+	holder_type = /obj/item/holder/rat/pet/ikit
+
+
+//Spawning those rats from cheese.
+
+/obj/item/reagent_container/food/snacks/cheesewedge/attack_self(mob/user)
+	if(!user.client || user.client.spawned_pet)
+		return ..()
+	if(GLOB.community_awards[user.ckey])
+		for(var/award in GLOB.community_awards[user.ckey])
+			if(award == "RatKing")
+				return spawn_pet_rat(user)
+	..()
+
+/client
+	/// Whether or not the client has spawned a unique pet.
+	var/spawned_pet = FALSE
```

Comment:
1 this shouldn't be a bespoke var, just use a trait
2 this shouldn't be on client because they can just reconnect to get a new client datum